packages/wonder-blocks-dropdown/components/dropdown-core.js

```diff
@@ -64,78 +74,175 @@ export default class DropdownCore extends React.Component<DropdownCoreProps> {
         document.removeEventListener("keyup", this._handleKeyup);
     }
 
     _handleInteract = (event: {target: any}) => {
-        if (this.node && this.node.contains(event.target)) {
-            return;
-        }
-        if (this.props.open) {
-            this.props.onClose();
+        const target: Node = event.target;
+        if (this.props.open && this.node && !this.node.contains(target)) {
+            this.props.onOpenChanged(false);
         }
     };
 
     _handleKeyup = (event: KeyboardEvent) => {
         if (this.props.open && event.key === "Escape") {
             event.preventDefault();
             event.stopImmediatePropagation();
-            this.props.onClose();
+            this.props.onOpenChanged(false);
         }
     };
 
+    getOpenerBounds() {
+        const {openerElement} = this.props;
+        if (openerElement) {
+            const rect = openerElement.getBoundingClientRect();
+            return {
+                top: rect.top + window.pageYOffset,
+                left: rect.left + window.pageXOffset,
+                width: rect.width,
+            };
+        }
+    }
+
+    getTranslatedPosition(openerBounds: {
+        top: number,
+        left: number,
+        width: number,
+    }) {
+        const {top, left, width} = openerBounds;
+        if (this.props.alignment === "left") {
+            return {
+                transform: `translate3d(${left}px, ${top}px, 0)`,
+            };
+        } else {
+            return {
+                transform: `translate3d(${left +
+                    width -
+                    (this.menuWidth || 0)}px, ${top}px, 0)`,
+            };
+        }
+    }
+
+    maybeRenderMenu() {
+        const {items, light, open, style} = this.props;
+        const body = document.getElementsByTagName("body");
+
+        const bounds = this.getOpenerBounds();
+        const translatedPosition = bounds && this.getTranslatedPosition(bounds);
+
+        if (open && body[0]) {
+            return ReactDOM.createPortal(
+                <View
+                    onMouseUp={(event) => {
+                        // Stop propagation to prevent the mouseup listener
+                        // on the document from closing the menu.
+                        event.nativeEvent.stopImmediatePropagation();
+                    }}
+                    ref={(node) =>
+                        (this.menuElement = ((ReactDOM.findDOMNode(
+                            node,
+                        ): any): Element))
+                    }
+                    style={[
+                        styles.dropdown,
+                        light && styles.light,
+                        !(open && this.menuWidth) && styles.hide,
+                        translatedPosition,
+                        style,
+                    ]}
+                >
+                    {items}
+                </View>,
+                body[0],
```

Comment:
I don't know where my comment went to about this.  I tried the test page out and it look like things are working correctly like this.  Maybe I'm misunderstanding the docs around needing a separate container.